### ISSUE TITLE
Subsequent AI Chat responses disappear on request finishes

### ISSUE THREAD

**Issue by SidneyCogdill** [Palm1r/QodeAssist#35]
It seems that only the first response of Chat can be properly displayed, the subsequent responses only show when receiving the token streams and disappear when the request finishes. The log output still has the full response though.

The OpenAI compatible API endpoint is used, if that matters. I don't have a clue where to start looking at because I can't run a debugger on it without pulling the whole Qt Creator source and building it with QodeAssist plugin.

**Comment by Palm1r** [comment]
For the chat I implemented a floating window mechanism, and if it exceeds the number of tokens, it should start deleting the original ones. Try expanding "Chat history token limit" it or try it on small messages. And if possible, send me a log, I will try to say the reason more precisely. What OpenAI compatible provider do you use?

**Comment by SidneyCogdill** [comment]
<details>
  <summary>Log</summary>

```
[Qode Assist] Sending request to llm: 
url: https://openrouter.ai/api/v1/chat/completions
Request body:
{
    "api_key": "---",
    "max_tokens": 2000,
    "messages": [
        {
            "content": "You are an advanced AI assistant specializing in C++, Qt, and QML development. Your role is to provide helpful, accurate, and detailed responses to questions about coding, debugging, and best practices in these technologies.",
            "role": "system"
        },
        {
            "content": "### Instruction:\nwrite a simple hello world (just the code)\n### Response:\n",
            "role": "user"
        }
    ],
    "model": "meta-llama/llama-3.1-70b-instruct:free",
    "stream": true,
    "temperature": 0.5
}

[Qode Assist] Message completed. Final response for message {38befe91-1dc6-4869-a22d-a471f8f2f577}: ``cpp
#include <iostream>

int main() {
    std::cout << "Hello, World!" << std::endl;
    return 0;
}
``
[Qode Assist] Message completed. Final response for message {38befe91-1dc6-4869-a22d-a471f8f2f577}: 
[Qode Assist] Request finished successfully
```

(yes, this output actually disappears in front of me. I tried to record it again but failed to reproduce with this prompt.)

</details>

The sliding window mechanism is probably not the issue here. A recording showcasing the bug:

https://github.com/user-attachments/assets/ed85f655-40ce-4ab5-af4a-7392c13e14b6

It seems to be easier to reproduce when asking CMake questions...?

**Comment by Palm1r** [comment]
Thank you, I will check today, I think I forgot something then moved to new settings

**Comment by SidneyCogdill** [comment]
BTW it might be feasible to decouple the AI Chat module from Qt Creator targets so that it can be independently tested by eg.. creating a new CMake executable target that builds a standalone Qt GUI AI Chat application. Currently debugging QodeAssist is kind of painful, I had to restart Qt Creator all the times.

**Comment by Palm1r** [comment]
Yes, I agree, it will be better for debugging. If you want, you can do it and I will upload it to the repository, or I will do it next weekend. P.S And sorry that the repository is not in the best shape, I wanted to finish embedding first and then start refactoring

**Comment by Palm1r** [comment]
sorry for delay, I rechecked, the problem with my templates for llama3 for openai api

**Comment by Palm1r** [comment]
@SidneyCogdill Problem is fixed, you can check it in version 0.3.10
